app/models/mixins/authentication_mixin.rb

```diff
@@ -79,7 +79,7 @@ def authentication_service_account(type = nil)
   end
 
   def required_credential_fields(_type)
-    [:userid]
+    _type.to_sym != :bearer ? [:userid] : []
```

Comment:
Also, prefer positive conditionals where possible...so in this case, `type.to_sym == :bearer ? [] : [:userid]` is generally considered easier to read.